OrbitQt/orbitlivefunctions.cpp

```diff
@@ -108,12 +108,14 @@ void OrbitLiveFunctions::Reset() {
   all_events_iterator_->DisableButtons();
 }
 
-void OrbitLiveFunctions::onRowSelected(int row) {
-  CHECK(row >= 0);
+void OrbitLiveFunctions::onRowSelected(std::optional<int> row) {
   ui->data_view_panel_->GetTreeView()->SetIsInternalRefresh(true);
   QItemSelectionModel* selection = ui->data_view_panel_->GetTreeView()->selectionModel();
-  QModelIndex idx = ui->data_view_panel_->GetTreeView()->GetModel()->CreateIndex(row, 0);
-  selection->select(idx, QItemSelectionModel::ClearAndSelect | QItemSelectionModel::Rows);
-  selection->setCurrentIndex(idx, QItemSelectionModel::Rows);
+  QModelIndex index;
+  if (row.has_value()) {
+    index = ui->data_view_panel_->GetTreeView()->GetModel()->CreateIndex(row.value(), 0);
+  }
+  selection->select(index, QItemSelectionModel::ClearAndSelect | QItemSelectionModel::Rows);
+  selection->setCurrentIndex(index, QItemSelectionModel::Rows);
```

Comment:
Are both calls above really needed? Looks a bit redundant, but I'm just guessing here.

Reply:
Also I'm assuming the default constructor of `QModelIndex` results in an index that will then clear the selection?

Reply:
> Also I'm assuming the default constructor of `QModelIndex` results in an index that will then clear the selection?

Yes. More specifically, when we deselect functions from the live tab or the capture window, we pass `std::nullopt` to `OnRowSelected(...)` to clear selections in the live tab.

Reply:
> Are both calls above really needed? Looks a bit redundant, but I'm just guessing here.

Thanks for pointing out this! Yes, it's redundant and I will fix it.